nwb_conversion_tools/datainterfaces/spikeglxdatainterface.py

```diff
@@ -9,6 +9,15 @@ class SpikeGLXRecordingInterface(BaseRecordingExtractorInterface):
 
     RX = SpikeGLXRecordingExtractor
 
+    def convert_data(self, nwbfile, metadata_dict: None, stub_test: bool = False, sync_with_ttl: bool = True,
+                     **conversion_options):
+        """Primary function for converting recording extractor data to nwb."""
+        if sync_with_ttl:
+            ttl, states = self.recording_extractor.get_ttl_events()
+            rising_times = ttl[states == 1]
+            self.recording_extractor = SubRecordingExtractor(self.recording_extractor, start_frame=rising_times[0])
```

Comment:
Could we move this to the `BaseRecordingExtractorInterface`?

Reply:
Unfortunately, no. The `.get_ttl_events()` is a function specific to the IntanRecordingExtractor.

Reply:
ah, ok. @alejoe91 would it be possible to implement that method more generally?

Reply:
Looks like get_ttl_events has been implemented for several RecordingExtractor types but not all of them. It looks like there was some method to this but it's not obvious to me. I'll resolve this comment, since this is a SpikeExtractors issue.

Reply:
OK; I'll keep it in mind though moving forward, if we get some dev refactor downtime away from conversion, this ought to at least in principle be generalizable.

Reply:
Actually it looks like `get_ttl_events` is implemented in the `RecordingExtractor` here: https://github.com/SpikeInterface/spikeextractors/blob/53b145e4b2d4f58ba34bcdcf6599352c2e7212cf/spikeextractors/recordingextractor.py#L768 so we should be able to call it more generally. It'll just raise a `NotImplementedError` if it isn't implemented for that subtype

Reply:
Ohh ok, I'll move this over then